codersdk/workspacesdk/workspacesdk.go

```diff
@@ -260,6 +260,19 @@ func (c *Client) DialAgent(dialCtx context.Context, agentID uuid.UUID, options *
 		options.Logger.Debug(ctx, "connected to tailnet v2+ API")
 	}
 
+	// TODO: uncomment after pgcoord ack's are implemented (upstack pr)
+	// options.Logger.Debug(ctx, "waiting for agent ack")
+	// // 5 seconds is chosen because this is the timeout for failed Wireguard
+	// // handshakes. In the worst case, we wait the same amount of time as a
+	// // failed handshake.
+	// timer := time.NewTimer(5 * time.Second)
+	// select {
+	// case <-connector.agentAck:
+	// case <-timer.C:
+	// 	options.Logger.Debug(ctx, "timed out waiting for agent ack")
+	// }
+	// timer.Stop()
+
```

Comment:
I don't think only guarding the keepalives will stop us from the 5s retry, we need to also stop the sdk from pinging.

Reply:
Hmm, should we just not add the node to tailscale on the client side until we get the ack?  I really would like to handle this inside the Conn instead of it leaking out to this layer because if we do it means we have to reimplement the behavior across coderd and wsproxies.